hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -405,7 +405,7 @@ private synchronized boolean getNamespaceEnabledInformationFromServer(
     }
     try {
       LOG.debug("Get root ACL status");
-      getClient().getAclStatus(AbfsHttpConstants.ROOT_PATH, tracingContext);
+      getClient(AbfsServiceType.DFS).getAclStatus(AbfsHttpConstants.ROOT_PATH, tracingContext);
```

Comment:
Why DFS client? 
In case of FNS, can't we just catch the UnsupportedOperationException and set/return isNamespaceEnabled to false?

Reply:
This piece of code is supposed to run during file system initialization. This is to determine the account type being used and this is supposed to run for all the combinations of account type and service endpoint configured. We do not want to use Blob Endpoint for any API on HNS enabled account and DFS Endpoint is the only endpoint supported for both FNS and HNS accounts. Hence, we make this call with DFS Endpoint.

Reply:
@anujmodi2021 I think an assertion could help here, something like below. Feel free to do it this PR or in another PR as this code already merged into trunk.

    Preconditions.checkArgument(uri.toString().contains(ABFS_DFS_DOMAIN_NAME), "<err_message>");

Reply:
So the method `getClient(AbfsServiceType.DFS)` will internally call `getClientHandler().getDfsClient()`.
The URi to interact with will be encapsulated into the client in this case DFS Client so the URL will always be DFS endpoit URL.

Refer to `AbfsClientHandler` constructor. There we explicitly check and make sure AbfsDfsClient is initialized with DFS Endpoint URL.

That's why I think this check here will be redundant.